internal/operators/metallb/manifest.go

```diff
@@ -0,0 +1,187 @@
+	openshiftManifests["50_openshift-metallb_operator_group.yaml"] = metalLBOperatorGroup
+	openshiftManifests["50_openshift-metallb_subscription.yaml"] = metalLBSubscription
+
+	return openshiftManifests, metalLBConfigs, nil
+}
+
+func getSubscription() ([]byte, error) {
+	data := map[string]string{
+		"OPERATOR_NAMESPACE":         Operator.Namespace,
+		"OPERATOR_SUBSCRIPTION_NAME": Operator.SubscriptionName,
+	}
+	return executeTemplate(data, "MetalLBSubscription", MetalLBSubscription)
+}
+
+func getNamespace() ([]byte, error) {
+	data := map[string]string{
+		"OPERATOR_NAMESPACE": Operator.Namespace,
+	}
+	return executeTemplate(data, "MetalLBNamespace", MetalLBNamespace)
+}
+
+func getOperatorGroup() ([]byte, error) {
+	data := map[string]string{
+		"OPERATOR_NAMESPACE":  Operator.Namespace,
+		"OPERATOR_GROUP_NAME": Operator.SubscriptionName,
+	}
+	return executeTemplate(data, "MetalLBOperatorGroup", MetalLBOperatorGroup)
+}
+
+func getMetalLBOperand(cluster *common.Cluster) ([]byte, error) {
+	properties, err := parsePropertiesField(cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	data := map[string]string{
+		"OPERATOR_NAMESPACE": Operator.Namespace,
+		"API_IP":             properties.ApiIP,
+		"INGRESS_IP":         properties.IngressIP,
+	}
+	return executeTemplate(data, "MetalLBOperand", MetalLBOperand)
+}
+
+func executeTemplate(data map[string]string, contentName, content string) ([]byte, error) {
+	tmpl, err := template.New(contentName).Parse(content)
+	if err != nil {
+		return nil, err
+	}
+	buf := &bytes.Buffer{}
+	err = tmpl.Execute(buf, data)
+	if err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+const MetalLBSubscription = `apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: "{{.OPERATOR_SUBSCRIPTION_NAME}}"
+  namespace: "{{.OPERATOR_NAMESPACE}}"
+spec:
+  channel: "stable"
+  name: metallb-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace`
+
+const MetalLBNamespace = `apiVersion: v1
+kind: Namespace
+metadata:
+  name: "{{.OPERATOR_NAMESPACE}}"
+  labels:
+    openshift.io/cluster-monitoring: "true"
+  annotations:
+    workload.openshift.io/allowed: management
+spec: {}`
+
+const MetalLBOperatorGroup = `apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: "{{.OPERATOR_GROUP_NAME}}"
+  namespace: "{{.OPERATOR_NAMESPACE}}"`
+
+const MetalLBOperand = `apiVersion: metallb.io/v1beta1
+kind: MetalLB
+metadata:
+  name: metallb
+  namespace: "{{.OPERATOR_NAMESPACE}}"
+---
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: metallb-api-vip
+  namespace: "{{.OPERATOR_NAMESPACE}}"
+spec:
+  autoAssign: false
+  addresses:
+    - {{.API_IP}}/32
+---
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: metallb-ingress-ip
+  namespace: "{{.OPERATOR_NAMESPACE}}"
+spec:
+  autoAssign: false
+  addresses:
+    - {{.INGRESS_IP}}/32
```

Comment:
Let's make sure these 2 are applied *only* if the IPs are provided.

I think it's fine to apply the following:

- L2Advertisement
- Services
- Operand

The service won't be fully configured and the L2Advertisement won't have an `IPAddressPool` to advertise until they are actually created upon relocation. This should be fine, may be worth testing to see what happens, tho.

Reply:
in case IPs are not provided we do not reach this point

Reply:
If the IPs are not provided we still need to deploy metallb if requested. We won't configure the `IPAddressPool` but we will configure the rest. Let's make sure this happens, pls